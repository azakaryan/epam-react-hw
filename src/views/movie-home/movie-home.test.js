import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import MovieHome from './movie-home';

expectEnzymify.extend(enzymify());
const onLoad = () => {};

describe('Movie Home Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MovieHome onLoad={onLoad} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a Movie Home component', () => {
    const wrapper = shallow(<MovieHome onLoad={onLoad} />);
    expectEnzymify(wrapper).toBeA('div');
  });

  it('find element', () => {
    let k = 1;
    const onLoad = () => k = k + 1;
    const wrapper = shallow(<MovieHome onLoad={onLoad} />);
    expect(k).toEqual(2);
    const el = wrapper.find({ id: 'movie-home' });
    expectEnzymify(el).toBeA('div');
  });
});
