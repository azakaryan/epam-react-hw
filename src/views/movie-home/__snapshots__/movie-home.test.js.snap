// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Home Component should render correctly 1`] = `
<div
  className="main_container"
  id="movie-home"
>
  <Header />
  <Connect(SearchFilters) />
  <div
    className="sort_and_count"
  >
    <Connect(MovieCount) />
    <Connect(SortFilters) />
  </div>
  <div>
    <Connect(MovieList) />
  </div>
</div>
`;
